core/src/main/java/org/apache/calcite/util/Sources.java

```diff
@@ -167,7 +167,7 @@ private FileSource(URL url) {
 
     private FileSource(File file) {
       this.file = Objects.requireNonNull(file);
-      this.url = null;
+      this.url = fileToUrl(file);
```

Comment:
This seems to impact `#toString()` output :-/

Reply:
Yes. It impacts the output of the toString method, when the source is constructed solely by a file object. 

I did not find any document specifying the output format, so maybe the client code should not depend on the specific format of the output.